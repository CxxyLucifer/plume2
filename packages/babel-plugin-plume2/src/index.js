/**
 * 判断当前的call-expression是不是require('react-dom')
 * @param {*} t
 * @param {*} path
 */

function isRequireReactDOM(t, path) {
  return (
    t.isIdentifier(path.node.callee, { name: 'require' }) &&
    path.get('arguments.0').isStringLiteral({ value: 'react-dom' })
  );
}

module.exports = function(babel) {
  var t = babel.types;

  return {
    visitor: {
      /**
       * if current env is react-native
       * import ReactDOM from 'react-dom' => import ReactDOM from 'react-native'
       * @param {*} path
       * @param {*} param1
       */
      ImportDeclaration(path, { opts }) {
        if (
          opts.reactnative &&
          t.isStringLiteral(path.node.source, {
            value: 'react-dom'
          })
        ) {
          path.node.source = t.StringLiteral('react-native');
        }
      },

      /**
       * if current env is react-native
       * var ReactDOM = require('react-dom') => var ReactDOM = require('react-native');
       * @param {*} path
       * @param {*} param1
       */ CallExpression(path, { opts }) {
        if (opts.reactnative && isRequireReactDOM(t, path)) {
          path.node.arguments[0] = t.StringLiteral('react-native');
        }
      }
    }
  };
};
