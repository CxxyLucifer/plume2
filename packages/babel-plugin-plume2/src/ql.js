/**
 * 判断当前的callexpress是不是QL
 * @param {*} path 
 */
export const isQLCallExpress = path => {
  const { node } = path;
  return node.callee.name === 'QL';
};

/**
 * 是不是可用的QL的语法
 * @param {*} path 
 */
export const isValidQLSyntax = path => {
  const { node } = path;
};
