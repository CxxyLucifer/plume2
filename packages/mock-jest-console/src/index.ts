declare const jest: { fn: Function };

export class BaseMock {
  constructor() {
    this._logs = [];
    this.mock();
  }

  private _logs;

  mock() {}

  mockFn() {
    return jest.fn(log => this._logs.push(log));
  }

  get logs() {
    return this._logs;
  }

  reset() {
    this._logs = [];
  }
}

export class MockWarn extends BaseMock {
  mock() {
    console.warn = this.mockFn();
  }
}

export class MockLog extends BaseMock {
  mock() {
    console.log = this.mockFn();
  }
}

export class MockDir extends BaseMock {
  mock() {
    console.dir = this.mockFn();
  }
}

export class MockDebug extends BaseMock {
  mock() {
    console.debug = this.mockFn();
  }
}

export class MockError extends BaseMock {
  mock() {
    console.error = this.mockFn();
  }
}

export class MockTrace extends BaseMock {
  mock() {
    console.trace = this.mockFn();
  }
}

export class MockGroup extends BaseMock {
  mock() {
    console.group = this.mockFn();
  }
}

export class MockGroupCollapsed extends BaseMock {
  mock() {
    console.groupCollapsed = this.mockFn();
  }
}

export class MockTime extends BaseMock {
  mock() {
    console.time = this.mockFn();
  }
}

export class MockConsole extends BaseMock {
  mock() {
    console.warn = this.mockFn();
    console.log = this.mockFn();
    console.dir = this.mockFn();
    console.debug = this.mockFn();
    console.error = this.mockFn();
    console.trace = this.mockFn();
    console.group = this.mockFn();
    console.groupCollapsed = this.mockFn();
    console.time = this.mockFn();
  }
}
