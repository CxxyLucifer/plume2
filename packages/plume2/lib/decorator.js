"use strict";
/**
 * 绑定Actor的Action
 * Usage:
 *  class HelloActor extends Actor {
 *     @Action('hello')
 *     hello(state) {
 *       return state;
 *     }
 *  }
 *
 * @param msg 事件名
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Action = function (msg) { return function (target, 
//@ts-ignore
property, descriptor) {
    target._route || (target._route = {});
    /**
     * 如果有actor的Action中有重名的事件名，warning
     */
    if (process.env.NODE_ENV != 'production') {
        if (target._route[msg]) {
            var actorName = target.constructor.name;
            console.warn("\uD83D\uDE0E" + actorName + " had @Action('" + msg + "'), Please review your code.");
        }
    }
    target._route[msg] = descriptor.value;
}; };
