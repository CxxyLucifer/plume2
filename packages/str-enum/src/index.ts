export default function StrEnum<T extends string>(
  ...args: Array<T>
): { [K in T]: K };

export default function StrEnum<
  T extends { [key: string]: V },
  V extends string
>(args: T): T;

export default function StrEnum(...args: Array<any>) {
  if (typeof args[0] === 'string') {
    const result = {} as any;
    for (let a of args) {
      result[a] = a;
    }
    return result;
  } else {
    return args[0];
  }
}
