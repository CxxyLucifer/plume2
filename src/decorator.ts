export function Action(msg: string): Function {
  return function(
    target: any,
    property: any,
    descriptor: TypedPropertyDescriptor<any>
  ) {
    target._route || (target._route = {});

    if (process.env.NODE_ENV != 'production') {
      if (target._route[msg]) {
        const actorName = target.constructor.name;
        console.warn(
          `ðŸ˜Ž${actorName} had @Action('${msg}'), Please review your code.`
        );
      }
    }

    target._route[msg] = descriptor.value;
  };
}
