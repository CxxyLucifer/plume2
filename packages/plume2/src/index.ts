import Actor from './actor';
import { Action } from './decorator';
import { go } from './go';
import msg from './msg';
import { PQL } from './pql';
import { QL } from './ql';
import Relax from './relax';
import Store from './store';
import StoreProvider from './store-provder';
import { IMap, IOptions, TViewAction } from './typing';
import { ViewAction } from './view-action';

export {
  go,
  QL,
  PQL,
  msg,
  Relax,
  ViewAction,
  Action,
  Actor,
  Store,
  StoreProvider,
  IMap,
  IOptions,
  TViewAction
};
