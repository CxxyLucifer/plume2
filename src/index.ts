import msg from './msg';
import { Action } from './decorator';
import Actor from './actor';
import Store from './store';
import StoreProvider from './store-provder';
import { QL } from './ql';
import Relax from './relax';

export { QL, msg, Relax, Action, Actor, Store, StoreProvider };
