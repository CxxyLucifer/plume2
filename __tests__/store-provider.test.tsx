import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Actor, Store, StoreProvider } from '../src/index';
import { Action } from '../src/decorator';

class HelloActor extends Actor {
  defaultState() {
    return { name: 'plume' };
  }

  @Action('change')
  change(state) {
    return state.set('name', 'plume++');
  }
}

class AppStore extends Store {
  constructor(props) {
    super(props);
    window['_store'] = this;
  }

  bindActor() {
    return [new HelloActor()];
  }

  change = () => {
    this.dispatch('change');
  };
}

@StoreProvider(AppStore)
class Home extends React.Component {
  state: { name: string };

  render() {
    return (
      <div>
        {this.state.name}
      </div>
    );
  }
}

describe('store provider test suite', () => {
  it('first render', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('store dispath change render', () => {
    const component = renderer.create(<Home />);
    const store = window['_store'] as AppStore;
    store.change();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
