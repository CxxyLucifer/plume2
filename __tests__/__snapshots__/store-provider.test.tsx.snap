// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store provider test suite first render 1`] = `
<div>
  plume
</div>
`;

exports[`store provider test suite store dispath change render 1`] = `
<div>
  plume++
</div>
`;
